src/io/flutter/run/bazelTest/FlutterBazelTestLineMarkerContributor.java

```diff
@@ -0,0 +1,116 @@
+ */
+package io.flutter.run.bazelTest;
+
+import com.intellij.execution.TestStateStorage;
+import com.intellij.execution.lineMarker.ExecutorAction;
+import com.intellij.execution.lineMarker.RunLineMarkerContributor;
+import com.intellij.execution.testframework.TestIconMapper;
+import com.intellij.execution.testframework.sm.runner.states.TestStateInfo;
+import com.intellij.icons.AllIcons;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiInvalidElementAccessException;
+import com.intellij.util.Function;
+import com.intellij.util.Time;
+import io.flutter.settings.FlutterSettings;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.Date;
+import java.util.Map;
+
+
+public class FlutterBazelTestLineMarkerContributor extends RunLineMarkerContributor {
+
+  private static final int SCANNED_TEST_RESULT_LIMIT = 1024;
+
+  @Nullable
+  @Override
+  public Info getInfo(@NotNull PsiElement element) {
+    final FlutterSettings settings = FlutterSettings.getInstance();
+    // Only the new bazel test runner supports running tests inside of a file or for a specific test name.
+    if (!settings.useNewBazelTestRunner(element.getProject())) {
+      return null;
+    }
+    final BazelTestConfigUtils.TestType testCall = BazelTestConfigUtils.asTestCall(element);
+    if (testCall != null) {
+      final Icon icon = getTestStateIcon(element, testCall.getIcon());
+      final Function<PsiElement, String> tooltipProvider =
+        psiElement -> testCall.getTooltip(psiElement);
+      return new Info(icon, tooltipProvider, ExecutorAction.getActions());
+    }
+
+    return null;
+  }
+
+  @NotNull
+  private static Icon getTestStateIcon(@NotNull PsiElement element, @NotNull Icon defaultIcon) {
+
+    // SMTTestProxy maps test run data to a URI derived from a location hint produced by `package:test`.
+    // If we can find corresponding data, we can provide state-aware icons. If not, we default to
+    // a standard Run state.
+
+    PsiFile containingFile;
+    try {
+      containingFile = element.getContainingFile();
+    }
+    catch (PsiInvalidElementAccessException e) {
+      containingFile = null;
+    }
+
+    final Project project = element.getProject();
+    final PsiDocumentManager psiDocumentManager = PsiDocumentManager.getInstance(project);
+
+    final Document document = containingFile == null ? null : psiDocumentManager.getDocument(containingFile);
+    if (document != null) {
+      final int textOffset = element.getTextOffset();
+      final int lineNumber = document.getLineNumber(textOffset);
+
+      // e.g., dart_location:///Users/pq/IdeaProjects/untitled1298891289891/test/unit_test.dart,3,2,["my first unit test"]
+      final String path = containingFile.getVirtualFile().getPath();
```

Comment:
Paths should be processed through FileUtil.toSystemIndependentName().  If such a call breaks some string assumption here, document the reason.